dispatcher.go

```diff
@@ -301,7 +301,7 @@ func (d *Dispatcher) Start() error {
 
 	abort := func(errs []error) error {
 		// Failed to start so stop everything that was started.
-		wait := intsync.ErrorWaiter{}
+		wait := errorsync.ErrorWaiter{}
```

Comment:
this is only used in the dispatcher, and it's implementation predates things like multierr, we should probably leave this for now since I don't expect encoding/transport authors to require this code.